File: backend/src/main/java/org/example/backend/domain/fish/dto/FishUpdateResponseDto.java
package org.example.backend.domain.fish.dto;

import java.time.LocalDateTime;
import org.example.backend.domain.fish.entity.Fish;

public record FishUpdateResponseDto(
    Long fishId,
    String fishSpecies,
    String fishName,
    LocalDateTime createdDate,
    LocalDateTime modifiedDate
) {

  public FishUpdateResponseDto(Fish fish) {
    this(
        fish.getId(),
        fish.getSpecies(),
        fish.getName(),
        fish.getCreateDate(),
        fish.getModifyDate()
    );
  }
}

File: backend/src/main/java/org/example/backend/domain/fish/dto/FishUpdateResponseDto.kt
package org.example.backend.domain.fish.dto

import org.example.backend.domain.fish.entity.Fish
import java.time.LocalDateTime

data class FishUpdateResponseDto(
    val fishId: Long,
    val fishSpecies: String,
    val fishName: String,
    val createdDate: LocalDateTime?,
    val modifiedDate: LocalDateTime?
) {
    constructor(fish: Fish) : this(
        fish.id,
        fish.species,
        fish.name,
        fish.createDate,
        fish.modifyDate
    )
}
